### PR TITLE
feat(sticky header utility): update storybook story & remove old dotcom-shell code

### LINKED ISSUE #8486: [Dotcom-Shell with TOC]: Story Grid alignment issues
**Engineering info:**  
- Innovation Team engineer: @andy-blum 
- Innovation Team JIRA Ticket: https://jsw.ibm.com/browse/HC-2430
- DPO consulting engineer: 

-----

### Description

- The issue is a follow-up to the Storybook gird alignment bugs that @oliviaflory noted in this PR (https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/8434#issuecomment-1065575789).
- Updates should be made in the `Dotcom-Shell without shell (fallback utility)` story

--------

_**Note:** All testing done on Chrome browser Version 98.0.4758.102 (Official Build) (x86_64)_

### Story grid alignment
I know we didn't have a design mock up for this story, but we do need to have everything aligned to the grid like the other Dotcom shell examples so we are demonstrating expected alignment to adopters.
Please let me know if the below examples are enough to go on!

**Vertical TOC stories**
Screen Shot 2022-03-11 at 4 57 48 PM

**Expecting** [Prod example link](https://www.ibm.com/standards/carbon/web-components/?path=/story/components-dotcom-shell--default)
Screen Shot 2022-03-11 at 4 47 04 PM

**Horizontal TOC Stories**
Screen Shot 2022-03-11 at 4 47 49 PM

**Expecting** [Prod example](https://www.ibm.com/standards/carbon/web-components/?path=/story/components-dotcom-shell--with-horizontal-toc)
Screen Shot 2022-03-11 at 4 46 09 PM

### Component(s) impacted

Dotcom-Shell without shell (fallback utility) 

### Browser

Chrome

### Carbon for IBM.com version

v.31.1

### Severity

Severity 3 = The problem is visible or noticeable to users but does not impede the usability or functionality. Affects minor functionality, has a workaround.

### Application/website

AEM

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

N/A

### Steps to reproduce the issue (if applicable)

_No response_

### Release date (if applicable)

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

### PR DESCRIPTION
**Pull request by andy-blum**
### Related Ticket(s)

Closes #8486 #8472

### Description

This PR is a followup top #8434 in which the dotcom-shell's scroll handling was mimicked in a utility class.

This PR aims to cleanup the storybook integration and removes now unnecessary code from the dotcom-shell-composite class.

### Changelog

**Changed**

- Updated storybook story for new dotcom-shell scrolling utility

**Removed**

- Scroll handling logic removed from the dotcom-shell component

### CODE REVIEW

**Comment by andy-blum** [comment]
I'm not sure what roadblocks I was hitting previously, but trashing my branch and starting fresh has yielded better results!

Remaining to-dos:
- [ ] Remove the `isNecessary` method from the utility & its usage within the WC components as it is now always necessary
- [ ] Refactor the `leadspace-with-search` component so that its sticky styles can match the dotcom-shell's functional specs
- [ ] Update the contents of the story to fully match the carbon grid

@ariellalgilmore you said you had a fix to the storybook story not matching the grid, I think?

**Comment by andy-blum** [comment]
Updated the styles in the leadspace-with-search component to better facilitate sticking to the top of the viewport.

There's currently an issue when the sticky search appears in which the page re-flow causes the browser to scroll up (but the page content stays steady). This causes the masthead to also jump down. On top of this, I still need to work in the search bar's slide-down transistion in a way that animates in when it should, but smoothly adjusts with the rest of the sticky headers and re-attaches to the the main document flow when scrolled all the way back to the top of the page.

File: packages/styles/scss/components/leadspace-with-search/_leadspace-with-search.scss [hunk 1]
```diff
@@ -285,13 +277,30 @@
...
-        transform: translate(-29%, $spacing-07);
+      .#{$prefix}--search-container {
+        @include carbon--make-row();
+
+        position: fixed;
+        z-index: 999;
+        width: 100%;
+        top: 0;
+        left: 0;
+        padding: 1rem;
```
Comment (kennylam) [inline packages/styles/scss/components/leadspace-with-search/_leadspace-with-search.scss:288]: Can use spacing token here.

**Review by kennylam — COMMENTED** [review]
Thanks @andy-blum! Just one thing I can see.

**Comment by proeung** [comment]
@andy-blum Looks like some e2e tests are failing (see - https://github.com/carbon-design-system/carbon-for-ibm-dotcom/runs/5875652807?check_suite_focus=true#step:8:4261). Can you take a look?

**Comment by ariellalgilmore** [comment]
Seeing two minor things when checking leadspace with search, but everything is looking great.

1: heading should be inside the gutter. Needs `padding-left: 1rem`
image

2. and seeing the animation happen slightly later. Should be 16px after the bottom of the search-with-typeahead
image

**Comment by andy-blum** [comment]
@ariellalgilmore alignment/sticky threshold should be fixed now

@proeung tests shouldn't be erroring anymore

**Comment by ariellalgilmore** [comment]
seeing this. Might need to check if masthead exists before setting cumulativeOffset? 
Screen Shot 2022-04-08 at 1 06 16 PM

File: packages/web-components/src/components/universal-banner/universal-banner.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ import settings from 'carbon-components/es/globals/js/settings';
 import ddsSettings from '../../internal/vendor/@carbon/ibmdotcom-utilities/utilities/settings/settings';
 import StableSelectorMixin from '../../globals/mixins/stable-selector';
 import styles from './universal-banner.scss';
-import StickyHeader from '../../internal/vendor/@carbon/ibmdotcom-utilities/utilities/StickyHeader/StickyHeader';
+import StickyHeader from '../../../../utilities/src/utilities/StickyHeader/StickyHeader.js';
```
Comment (ariellalgilmore) [inline packages/web-components/src/components/universal-banner/universal-banner.ts:17]: wondering if we can change these imports back? the Web components HTML tests are failing

**Review by ariellalgilmore — APPROVED** [review]
LGTM! Thanks @andy-blum!!

**Review by proeung — APPROVED** [review]
LGTM!

Also, we'll need some design QA review before this PR is ready to be merged down. @oliviaflory or @RichKummer When you have a moment, can you help test this PR? Thanks!

**Comment by andy-blum** [comment]
Latest change is not just a stylefix, it adds a new DOM element. @ariellalgilmore @proeung @RichKummer please re-review when you have a few minutes

**Review by RichKummer — APPROVED** [review]
Thanks for fixing @andy-blum ! LGTM 🚀

**Review by proeung — APPROVED** [review]
LGTM!

**Pull request merged**